perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/GroupsManagerMethod.java

```diff
@@ -1759,5 +1758,78 @@ public Member call(ApiCaller ac, Deserializer parms) throws PerunException {
 					ac.getGroupById(parms.readInt("group")),
 					status);
 		}
+	},
+
+	/*#
+	 * Returns all groups which can be registered into during vo registration.
+	 *
+	 * @throw VoNotExistsException When the vo doesn't exist
+	 *
+	 * @param vo int vo <code>id</code>
+	 * @return List<Group> list of groups
+	 */
+	getGroupsToAutoRegistration {
+		@Override
+		public List<Group> call(ApiCaller ac, Deserializer parms) throws PerunException {
+			return ac.getGroupsManager().getGroupsForAutoRegistration(ac.getSession(),
+				ac.getVoById(parms.readInt("vo")));
+		}
+	},
+
+	/*#
+	 * Deletes groups from a list of groups which can be registered into during vo registration.
+	 *
+	 * @throw GroupNotExistsException When the group doesn't exist
+	 * @throw VoNotExistsException When the vo doesn't exist
+	 * @throw GroupAlreadyRemovedException When deleting group which is already removed
+	 *
+	 * @param vo int vo <code>id</code>
+	 * @param ids List<Integer> list of groups IDs
+	 */
+	deleteGroupsToAutoRegistration {
+		@Override
+		public List<Group> call(ApiCaller ac, Deserializer parms) throws PerunException {
+			parms.stateChangingCheck();
+			List<Integer> groupsInts = parms.readList("groups", Integer.class);
+
+			List<Group> groups = new ArrayList<>();
+			for (Integer groupInt : groupsInts) {
+				groups.add(ac.getGroupById(groupInt));
+			}
+
+			ac.getGroupsManager().deleteGroupsFromAutoRegistration(ac.getSession(),
+				ac.getVoById(parms.readInt("vo")),
```

Comment:
It shouldn't be necessary to specify vo in this method since each group has exactly one vo assigned, therefore, we can get it ourselves.

Reply:
Ok, I did not realise that, it should be removed.